__init__.py

```diff
@@ -2678,699 +3035,720 @@ def get_eigenvalues(self, kpt=None, spin=None, efermi=None):
+            eig.append(Hartree * np.fromfile(
+                f, dtype=float, count=nbnd, sep=' ') - ef)
             f.close()
 
-        spinall = spin not in ('up','down',0,1)
+        spinall = spin not in ('up', 'down', 0, 1)
         if kpt is not None and spinall:
             return np.array(eig[0])
         elif kpt is None and spinall and self.spinpol:
-            return np.reshape(np.array(eig), (2,nkp2,nbnd))
+            return np.reshape(np.array(eig), (2, nkp2, nbnd))
         else:
             return np.array(eig)
 
-
-
     def read_3d_grid(self, stream, log):
-        f = open(self.localtmp+'/'+log, 'a')
-        x = stream.readline()
-        while x!='' and x[:11]!='DATAGRID_3D':
-            f.write(x)
-            x = stream.readline()
-        if x=='':
-            raise RuntimeError, 'error reading 3D data grid'
-        f.write(x)
-        nx, ny, nz = [int(y) for y in stream.readline().split()]
-        origin = np.array([float(y) for y in stream.readline().split()])
-        cell = np.empty((3,3), np.float)
-        for i in range(3):
-            cell[i][:] = [float(y) for y in stream.readline().split()]
-        data = np.reshape(np.fromfile(stream, count=nx*ny*nz, sep=' '),
-            (nx,ny,nz), order='F')
-
-        x = stream.readline()
-        while x!='':
+        with open(self.localtmp + '/' + log, 'a') as f:
+            x = stream.readline().decode('utf-8')
+            while x != '' and x[:11] != 'DATAGRID_3D':
+                f.write(x)
+                x = stream.readline().decode('utf-8')
+            if x == '':
+                raise RuntimeError('error reading 3D data grid')
             f.write(x)
-            x = stream.readline()
-
-        f.close()
-        return (origin,cell,data)
 
+            n = [int(_) for _ in stream.readline().split()]
+            origin = np.array([float(_) for _ in stream.readline().split()])
 
-    def read_2d_grid(self, stream, log):
-        f = open(self.localtmp+'/'+log, 'a')
-        x = stream.readline()
-        while x!='' and x[:11]!='DATAGRID_2D':
-            f.write(x)
-            x = stream.readline()
-        if x=='':
-            raise RuntimeError, 'error reading 2D data grid'
-        f.write(x)
-        nx, ny = [int(y) for y in stream.readline().split()]
-        origin = np.array([float(y) for y in stream.readline().split()])
-        cell = np.empty((3,3), np.float)
-        for i in range(3):
-            cell[i][:] = [float(y) for y in stream.readline().split()]
-        data = np.reshape(np.fromfile(stream, count=nx*ny, sep=' '),
-            (nx,ny), order='F')
+            cell = np.empty((3, 3))
+            for i in range(3):
+                cell[i][:] = [float(_) for _ in stream.readline().split()]
 
-        x = stream.readline()
-        while x!='':
-            f.write(x)
-            x = stream.readline()
+            data = []
+            ntotal = np.prod(n)
+            while len(data) < ntotal:
```

Comment:
Change from np.fromfile to loop which accomplishes the same effect. The bytes stdout from the subprocess command was not being handled correctly.